codegen/src/utils/fetch_metadata.rs

```diff
@@ -46,15 +46,15 @@ impl std::str::FromStr for MetadataVersion {
 
 /// Returns the metadata bytes from the provided URL, blocking the current thread.
 pub fn fetch_metadata_bytes_blocking(
-    url: &Uri,
+    url: Url,
     version: MetadataVersion,
 ) -> Result<Vec<u8>, FetchMetadataError> {
     tokio_block_on(fetch_metadata_bytes(url, version))
 }
 
 /// Returns the raw, 0x prefixed metadata hex from the provided URL, blocking the current thread.
 pub fn fetch_metadata_hex_blocking(
-    url: &Uri,
+    url: Url,
```

Comment:
I wonder why passing the reference no longer works with `Url` but did with `Uri`? (this trickled up to meaning ectra clones and stuff in other places; no big deal but curious :))

Reply:
Yeah, if look at by comment above you can see that we are parsing an "already parsed URL" using its String representation :D